lib/tools/apk-utils.js

```diff
@@ -473,13 +473,79 @@ apkUtilsMethods.getDeviceLocale = async function () {
 };
 
 /**
- * Set the locale name of the device under test.
+ * Set the locale name of the device under test and the format of the locale is en-US, for example.
+ * This method call setDeviceLanguageCountry, so, please use setDeviceLanguageCountry as possible.
  *
- * @param {string} locale - The name of the new device country.
+ * @param {string} locale - The name of device country connected with `-`. e.g. en-US.
  */
 apkUtilsMethods.setDeviceLocale = async function (locale) {
-  // this method is only used in API >= 23
-  await this.setDeviceSysLocale(locale);
+  let split_locale = locale.split("-");
+  await this.setDeviceLanguageCountry(split_locale[0], split_locale[1]);
+};
+
+/**
+ * Set the locale name of the device under test.
+ *
+ * @param {string} language - Language. format: [a-zA-Z]{2,8}. e.g. en, ja : https://developer.android.com/reference/java/util/Locale.html
+ * @param {string} country - Country. format: [a-zA-Z]{2} | [0-9]{3}. e.g. US, JP : https://developer.android.com/reference/java/util/Locale.html
+ */
+apkUtilsMethods.setDeviceLanguageCountry = async function (language, country) {
+  let hasLanguage = language && _.isString(language);
+  let hasCountry = country && _.isString(country);
+  if (!hasLanguage && !hasCountry) {
+    log.debug(`setDeviceLanguageCountry requires language or country.`);
+    log.debug(`language: '${language}' and country: '${country}'`);
+    return;
+  }
+  let wasSettingChanged = false;
+  let apiLevel = await this.getApiLevel();
+
+  if (apiLevel < 23) {
+    let curLanguage = await this.getDeviceLanguage();
+    let curCountry = await this.getDeviceCountry();
+    if (hasLanguage && language !== curLanguage) {
+      await this.setDeviceLanguage(language);
+      wasSettingChanged = true;
+    }
+    if (hasCountry && country !== curCountry) {
+      await this.setDeviceCountry(country);
+      wasSettingChanged = true;
+    }
+  } else {
+    let curLocale = await this.getDeviceLocale();
+
+    if (apiLevel === 23) {
+      let locale;
+      if (!hasCountry) {
+        locale = language;
+      } else if (!hasLanguage) {
+        locale = country;
+      } else {
+        locale = `${language}-${country}`;
+      }
+
+      log.debug(`Current locale: '${curLocale}'; requested locale: '${locale}'`);
+      if (locale !== curLocale) {
+        await this.setDeviceSysLocale(locale);
+        wasSettingChanged = true;
+      }
+    } else { // API >= 24
+      if (!hasCountry || !hasLanguage) {
+        log.debug(`setDeviceLanguageCountry requires both language and country to be set for API 24+`);
```

Comment:
I'd rather change the level to `warn` and add a mention about device locale is not going to be changed